lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,26 @@
+module Semian
+  class AtomicInteger < SharedMemoryObject #:nodoc:
+
+    def initialize(name, permissions)
+      data_layout = [:int]
+      if acquire(name, data_layout, permissions)
+      else
+        @value=0
+      end
```

Comment:
It's unclear to me what's going on here. You attempt to acquire the data structure, and if you don't, you set `@value` to `0`. So what happens when I call `#value` and it acquired the data structure?